gax-java/gax/src/test/java/com/google/api/gax/rpc/EndpointContextTest.java

```diff
@@ -344,7 +339,7 @@ public void endpointContextBuild_gdchFlow_noUniverseDomain_customEndpoint() thro
   // For this test running locally or in CI, check that the Env Var is set properly.
   // This test should only run when the maven profile `EnvVarTest` is enabled.
   @Test
-  public void endpointContextBuild_universeDomainEnvVarSet() throws IOException {
+  void endpointContextBuild_universeDomainEnvVarSet() throws IOException {
```

Comment:
If possible, wan we convert this test to use JUnit5 functionality to set the Env Var inside the test? 

We can remove the custom configs: https://github.com/googleapis/sdk-platform-java/blob/94ef4f42cfda9300f60acb9e6c14e0ca02109716/gax-java/gax/pom.xml#L114-L129
and https://github.com/googleapis/sdk-platform-java/blob/94ef4f42cfda9300f60acb9e6c14e0ca02109716/gax-java/gax/pom.xml#L108

as well as the CI job:
- https://github.com/googleapis/sdk-platform-java/blob/94ef4f42cfda9300f60acb9e6c14e0ca02109716/.github/workflows/ci.yaml#L26-L29
- https://github.com/googleapis/sdk-platform-java/blob/94ef4f42cfda9300f60acb9e6c14e0ca02109716/.github/workflows/ci.yaml#L76-L79
-https://github.com/googleapis/sdk-platform-java/blob/94ef4f42cfda9300f60acb9e6c14e0ca02109716/.github/workflows/ci.yaml#L117-L129

Reply:
I listed them as out of scope of this PR in the description, as the main concern of this PR is to remove all Junit 4 usages. I'll create an issue for it and we can address it in a follow up PR.

Reply:
Ok, sounds good.